server_tgi.py

```diff
@@ -0,0 +1,233 @@
+import argparse
+import time
+import uuid
+from contextlib import asynccontextmanager
+
+import uvicorn
+from fastapi import FastAPI, Request
+from fastapi.responses import StreamingResponse
+from huggingface_hub import InferenceClient
+from transformers import AutoTokenizer
+
+from functionary.inference_stream import generate_openai_format_from_stream_async
+from functionary.openai_types import (
+    ChatCompletionChunk,
+    ChatCompletionRequest,
+    ChatCompletionResponse,
+    ChatCompletionResponseChoice,
+    ChatMessage,
+    Function,
+    StreamChoice,
+    Tool,
+    UsageInfo,
+)
+from functionary.prompt_template import get_prompt_template_from_tokenizer
+
+# @asynccontextmanager
+# async def lifespan(app: FastAPI):
+#     # Run TGI docker container at start up
+#     tgi_docker_client = docker.from_env()
+#     tgi_container = tgi_docker_client.containers.run(
+#         "ghcr.io/huggingface/text-generation-inference:2.0.4",
+#         f"--model-id {args.model} --max-batch-prefill-tokens 8242 --max-total-tokens 8192 --max-input-tokens 8191",
+#         device_requests=[docker.types.DeviceRequest(count=-1, capabilities=[["gpu"]])],
+#         shm_size="1g",
+#         ports={"8080/tcp": 80},
+#         volumes={"/home/jeffrey/data": {"bind": "/data", "mode": "rw"}},
+#         detach=True,
+#     )
+#     yield
+#     # Stop and remove TGI docker container at shutdown
+#     tgi_container.stop()
+#     tgi_container.remove()
+
+
+# docker run --gpus all --shm-size 1g -p 8080:80 -v $volume:/data \
+#     ghcr.io/huggingface/text-generation-inference:2.0.4 \
+#     --model-id $model --max-batch-prefill-tokens 8242 --max-total-tokens 8192 --max-input-tokens 8191
+
+
+app = FastAPI(title="Functionary TGI")  # , lifespan=lifespan)
+
+
+@app.post("/v1/chat/completions")
+async def create_chat_completion(raw_request: Request):
```

Comment:
this raw_request can be a dict. So you dont need to do `await raw_request.json()`. fastapi will automatically convert